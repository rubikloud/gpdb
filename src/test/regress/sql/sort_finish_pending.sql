drop table if exists _tmp_table;
create table _tmp_table (i1 int, i2 int, i3 int, i4 int);
insert into _tmp_table select i, i % 100, i % 10000, i % 75 from generate_series(0,99999) i;

-- make sort to spill
set statement_mem="2MB";
set gp_enable_mk_sort=on;
set gp_cte_sharing=on;


--start_ignore
\! gpfaultinjector -f execsort_mksort_mergeruns -y reset --seg_dbid 2
-- set QueryFinishPending=true in sort mergeruns. This will stop sort and set result_tape to NULL
\! gpfaultinjector -f execsort_mksort_mergeruns -y finish_pending --seg_dbid 2
--end_ignore

-- return results although sort will be interrupted in one of the segments 
select DISTINCT S from (select row_number() over(partition by i1 order by i2) AS T, count(*) over (partition by i1) AS S from _tmp_table) AS TMP;
