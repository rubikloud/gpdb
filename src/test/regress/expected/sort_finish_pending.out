drop table if exists _tmp_table;
NOTICE:  table "_tmp_table" does not exist, skipping
create table _tmp_table (i1 int, i2 int, i3 int, i4 int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i1' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
insert into _tmp_table select i, i % 100, i % 10000, i % 75 from generate_series(0,99999) i;
-- make sort to spill
set statement_mem="2MB";
set gp_enable_mk_sort=on;
set gp_cte_sharing=on;
--start_ignore
\! gpfaultinjector -f execsort_mksort_mergeruns -y reset --seg_dbid 2
20160114:18:51:50:053424 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f execsort_mksort_mergeruns -y reset --seg_dbid 2
20160114:18:51:50:053424 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-local Greenplum Version: 'postgres (Greenplum Database) 4.3.99.00 build dev'
20160114:18:51:50:053424 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Obtaining Segment details from master...
20160114:18:51:50:053424 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on 1 segment(s)
20160114:18:51:50:053424 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on nikos-mac:/Users/narmenatzoglou/gitdev/gpdb/gpdb-data/p1/gpseg0:content=0:dbid=2:mode=s:status=u
20160114:18:51:50:053424 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
-- set QueryFinishPending=true in sort mergeruns. This will stop sort and set result_tape to NULL
\! gpfaultinjector -f execsort_mksort_mergeruns -y finish_pending --seg_dbid 2
20160114:18:51:50:053436 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Starting gpfaultinjector with args: -f execsort_mksort_mergeruns -y finish_pending --seg_dbid 2
20160114:18:51:51:053436 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-local Greenplum Version: 'postgres (Greenplum Database) 4.3.99.00 build dev'
20160114:18:51:51:053436 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Obtaining Segment details from master...
20160114:18:51:51:053436 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on 1 segment(s)
20160114:18:51:51:053436 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-Injecting fault on nikos-mac:/Users/narmenatzoglou/gitdev/gpdb/gpdb-data/p1/gpseg0:content=0:dbid=2:mode=s:status=u
20160114:18:51:51:053436 gpfaultinjector:nikos-mac:narmenatzoglou-[INFO]:-DONE
--end_ignore
-- return results although sort will be interrupted in one of the segments 
select DISTINCT S from (select row_number() over(partition by i1 order by i2) AS T, count(*) over (partition by i1) AS S from _tmp_table) AS TMP;
 s 
---
 1
(1 row)

